packages/dbos-cloud/applications/deploy-app-code.ts

```diff
@@ -147,10 +161,16 @@ export async function deployAppCode(host: string, rollback: boolean): Promise<nu
   }
 }
 
-function readInterpolatedConfig(configFilePath: string): string {
+function readInterpolatedConfig(configFilePath: string, verbose: boolean, logger: {info: (arg: string)=>void}): string {
   const configFileContent = checkReadFile(configFilePath) as string;
   const regex = /\${([^}]+)}/g;  // Regex to match ${VAR_NAME} style placeholders
   return configFileContent.replace(regex, (_, g1: string) => {
-    return process.env[g1] || "";  // If the env variable is not set, return an empty string.
-});
+    if (process.env[g1] !== undefined) {
+      if (verbose) logger.info(`      Inserting value of '${g1}' from process environment.`);
+      return process.env[g1] ?? "";
+    }
+
+    if (verbose) logger.info(`      Variable '${g1}' would be taken from process environment, but is not defined.`);
```

Comment:
Ditto, "substituted" instead of "taken"

Reply:
Yeah the current wording didn't seem quite right, I'll change it.